portal/cache.py

```diff
@@ -1,9 +1,22 @@
+from flask import request, url_for
 from flask_caching import Cache
 
-
-# Simple init, replaced during app configuration
-cache = Cache(config={'CACHE_TYPE': 'simple'})
-
+# Configured during app configuration
+cache = Cache()
 
 # Human readable constants, values in seconds, for cache timeouts
 TWO_HOURS = 2*60*60
+FIVE_MINS = 5*60
+
+
+def request_args_in_key():
+    """Set as `key_prefix=request_args_in_key` for safe view caching
+
+    By default, cache.cached() uses only the request path as the key.
+    For any views returning different results with query string parameters,
+    need to include those parameters in the cache key
```

Comment:
@achen2401 FYI we may need to remove `cache: false` from some jQuery AJAX calls for this to take effect.

As I understand it, we should generally be able to rely on the browser doing the "right thing" with caching.

Our static content has an 8 hour `Expires` header, but otherwise we do not add any headers to instruct the browser to cache requests, so it should make a fresh request every time.

---
See [`jQuery.ajax()`](https://api.jquery.com/jquery.ajax/)
```
It works by appending "_={timestamp}" to the GET parameters
```

Reply:
@achen2401 an especially good one to test is any calls you make to /api/organization - there should be a notable difference in timing when getting the cached version.